core/src/Lender.sol

```diff
@@ -235,7 +237,12 @@ contract Lender is Ledger {
         unchecked {
             // Convert `amount` to `units`
             units = (amount * BORROWS_SCALER) / cache.borrowIndex;
-            require(units < b, "Aloe: repay too much");
+            if (!(units < b)) {
+                units = b - 1;
+
+                uint256 maxRepay = (units * cache.borrowIndex).unsafeDivUp(BORROWS_SCALER);
+                require(b > 1 && amount <= maxRepay, "Aloe: repay too much");
+            }
```

Comment:
`b - 1` is the maximum value we can subtract from the `borrows` mapping because there needs to at least 1 unit left as a flag that indicates the user is, in fact, a borrower.

However we can't simply clamp to that in all cases. Imagine a user sends $1000000 to the contract and calls this function with that amount, thinking they have that much debt. But in reality they have only $100 of debt. If we clamped to `b - 1` in all cases, `repay` would succeed and the remaining $999900 would be wasted (and likely stolen).

To protect users, we ensure the `amount <= maxRepay`. This is the same sort of protection we had before this PR, but now it's compatible with the whole "If you pay off `borrowBalance` you're guaranteed to be left with 0 debt" thing.